### PR TITLE
Mobile direct messages view

### PR DESCRIPTION
**Pull request by mxstbr**
**Status**
- [x] WIP
- [ ] Ready for review
- [ ] Needs testing

**TODO**

- [x] Render list of DM threads
- [x] Navigate to individual DM thread screen on click
- [x] Show messages on individual DM thread screen
- [x] Show chat input on individual DM thread screen
- [x] Send DMs from mobile
- [x] Make list of DM threads look nice
  - [x] Participant heads
  - [x] Proper list items
- [x] Make individual DM thread screen look nice
  - [x] Participant heads
- [x] Switch `<Messages />` to use `<InfiniteList />`
- [ ] Fix sending quotes in DM threads after merging #3088 

**Current status**

Recording of current status

### CODE REVIEW

File: mobile/components/Flex/Row.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// @flow
+import styled from 'styled-components/native';
+
+export default styled.View`
+  flex-direction: row;
```
Comment (mxstbr) [inline mobile/components/Flex/Row.js:5]: - Added `<Row />` and `<Column />` components

File: mobile/views/TabBar/DirectMessageStack.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// @flow
+import * as React from 'react';
+import { Button } from 'react-native';
+import { StackNavigator } from 'react-navigation';
+import idx from 'idx';
```
Comment (mxstbr) [inline mobile/views/TabBar/DirectMessageStack.js:5]: - Added `idx` to the deps and used it throughout, shoutout to @alejandronanez for the tip

File: shared/time-difference.js [hunk 3]
```diff
@@ -1,35 +1,79 @@
...
+  } else if (elapsed < MS_PER_DAY) {
+    return Math.round(elapsed / MS_PER_HOUR) + 'h';
+  } else if (elapsed < MS_PER_YEAR) {
+    return Math.round(elapsed / MS_PER_DAY) + 'd';
+  } else {
+    return Math.round(elapsed / MS_PER_YEAR) + 'y';
+  }
+}
+
+export function timeDifference(current: number, previous: ?number): string {
```
Comment (mxstbr) [inline shared/time-difference.js:24]: - Moved these to `shared`, removed duplicates from `src`

**Comment by mxstbr** [comment]
@brianlovin this is pretty much ready for review! 😍  Note that this doesn't have the chat input/messages enhancements from #3088, so don't expect the messages screen to look that great yet. (marked some important-ish points inline)

**Review by brianlovin — APPROVED** [review]
LGTM! Architecture makes sense, will dig in locally and help tidy up things as well

File: mobile/views/TabBar/DirectMessageStack.js (hunk 2 above)
Comment (alejandronanez) [reply mobile/views/TabBar/DirectMessageStack.js:5]: Awesome!

**Pull request merged**